src/App.tsx

```diff
@@ -1,24 +1,150 @@
+import { Todo } from './types/Todo';
+import { Filters } from './utils/Filters';
+import { NewTodo } from './components/NewTodo/NewTodo';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import {
+  deleteTodo,
+  getTodos,
+} from './api/todos';
+import { countTodos } from './utils/countTodos';
+import {
+  useTodos,
+} from './components/Contexts/TodosContext';
+import { USER_ID } from './utils/userToken';
+import {
+  useLoadingTodos,
+} from './components/Contexts/LoadingTodosContext';
+import {
+  useErrorMessage,
+} from './components/Contexts/ErrorMessageContext';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const { todos, setTodos } = useTodos();
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const { setLoadingTodos } = useLoadingTodos();
+  const {
+    errorMessage,
+    setErrorMessage,
+    isErrorHidden,
+    setIsErrorHidden,
+  } = useErrorMessage();
+
+  const [filterParam, setFilterParam] = useState(Filters.All);
+
+  const clearCompletedTodos = () => {
+    const completedTodos = countTodos(todos, true);
+
+    completedTodos.forEach(({ id }) => {
+      setLoadingTodos(prev => [...prev, { todoId: id, isLoading: true }]);
+      deleteTodo(id)
+        .then(() => {
+          setTodos(prev => prev.filter((todo) => id !== todo.id));
+        })
+        .catch((error) => {
+          setErrorMessage(JSON.parse(error.message).error);
+          setIsErrorHidden(false);
+
+          setTimeout(() => {
+            setIsErrorHidden(true);
+          }, 3000);
+        })
+        .finally(() => {
+          setLoadingTodos(prev => prev.filter(todo => todo.todoId !== id));
+        });
+    });
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch((error) => {
+        setErrorMessage(JSON.parse(error.message).error);
+        setIsErrorHidden(false);
+
+        setTimeout(() => {
+          setIsErrorHidden(true);
+        }, 3000);
+      });
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    switch (filterParam) {
+      case Filters.Active:
+        return todos.filter(({ completed }) => !completed);
+
+      case Filters.Completed:
+        return todos.filter(({ completed }) => completed);
+
+      case Filters.All:
+        return todos;
+
+      default:
+        return todos;
+    }
+  }, [filterParam, todos]);
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <NewTodo
+          onWaiting={setTempTodo}
+        />
```

Comment:
```suggestion
        <NewTodo onWaiting={setTempTodo} />
```
Better write it in one line if there are < 3 props